pkg/stanza/fileconsumer/splitter_factory.go

```diff
@@ -57,3 +59,39 @@ func (factory *multilineSplitterFactory) Build(maxLogSize int) (bufio.SplitFunc,
 	}
 	return splitter, nil
 }
+
+type defaultSplitterFactory struct {
+	EncodingConfig helper.EncodingConfig
+	Flusher        helper.FlusherConfig
+}
+
+var _ splitterFactory = (*defaultSplitterFactory)(nil)
+
+func newDefaultSplitterFactory(
+	encoding helper.EncodingConfig,
+	flusher helper.FlusherConfig) *defaultSplitterFactory {
+	return &defaultSplitterFactory{
+		EncodingConfig: encoding,
+		Flusher:        flusher,
+	}
+}
+
+// Build builds default Splitter struct
+func (factory *defaultSplitterFactory) Build(maxLogSize int) (bufio.SplitFunc, error) {
+	enc, err := factory.EncodingConfig.Build()
+	if err != nil {
+		return nil, err
+	}
+	if enc.Encoding == encoding.Nop {
+		return helper.SplitNone(maxLogSize), nil
+	}
+	splitFunc, err := helper.NewNewlineSplitFunc(enc.Encoding, false)
+	if err != nil {
+		return nil, err
+	}
+	flusher := factory.Flusher.Build()
+	if flusher != nil {
+		splitFunc = flusher.SplitFunc(splitFunc)
+	}
```

Comment:
I think the default factory should not do this, but a flusher can be added with a factory option.

Reply:
`flusherConfig.Build()` returns a pointer. If flusher is added with a factory option, it will cause data race.
Is it reasonable for us to create a new flusher pointer in the factory?

Reply:
flusherConfig.Build() returns a pointer. If flusher pointer is added with a factory option, it will cause data race.
Can we pass `flusherConfig` as an option to the factory and then build a `flusher` pointer when building the factory?

Reply:
Seems reasonable

Reply:
I modified the implementation of the code.`splitterfactory.Build` passes in `encoding.Encoding` and `helper.Flusher` instead of building them.
If `helper.Flusher` is passed in when constructing the factory class, it will cause  data race. So, I choose to pass in `helper.Flusher` when calling the `splitterfactory.Build`.